src/projection.js

```diff
@@ -232,3 +231,17 @@ function applyProjection(cx, cy, values, projection) {
     stream.point(x[i], y[i]);
   }
 }
+
+// When a named projection is specified, we can use its natural aspect ratio to
+// determine a good value for the projection’s height based on the desired
+// width. When we don’t have a way to know, the golden ratio is our best guess.
+export function projectionAspectRatio(projection, geometry) {
+  if (isObject(projection)) projection = projection.type;
```

Comment:
This is still not quite right because in Projection(…) we check typeof projection.stream. We need to make sure that this code interprets the projection option in _exactly_ the same way so that we don’t have differing definitions on whether a projection has been specified.

(Although, some difference is unavoidable because it’s possible for the projection initializer to be a function that then returns a null projection; we would still consider that a non-null projection for the purposes of automatic height. But that’s such an edge case I’m not worried about it.)